src/renderer/modals/SellDeviceConfirm/fields/AmountField.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import React, { useCallback, useEffect } from "react";
+import { BigNumber } from "bignumber.js";
+import { Trans } from "react-i18next";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge";
+import type {
+  Account,
+  AccountLike,
+  Transaction,
+  TransactionStatus,
+} from "@ledgerhq/live-common/lib/types";
+import type { TFunction } from "react-i18next";
+
+import Box from "~/renderer/components/Box";
+import Label from "~/renderer/components/Label";
+import RequestAmount from "~/renderer/components/RequestAmount";
+import Switch from "~/renderer/components/Switch";
+import Text from "~/renderer/components/Text";
+
+type Props = {
+  parentAccount: ?Account,
+  account: AccountLike,
+  transaction: Transaction,
+  onChangeTransaction: (*) => void,
+  status: TransactionStatus,
+  bridgePending: boolean,
+  t: TFunction,
+  initValue?: BigNumber,
+  resetInitValue?: () => void,
+};
+
+const AmountField = ({
+  account,
+  parentAccount,
+  transaction,
+  onChangeTransaction,
+  status,
+  bridgePending,
+  t,
+  initValue,
+  resetInitValue,
+}: Props) => {
+  const bridge = getAccountBridge(account, parentAccount);
+
+  useEffect(() => {
+    if (initValue && !initValue.eq(transaction.amount || BigNumber(0))) {
+      onChangeTransaction(bridge.updateTransaction(transaction, { amount: initValue }));
+      resetInitValue && resetInitValue();
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const onChange = useCallback(
+    (amount: BigNumber) => {
+      onChangeTransaction(bridge.updateTransaction(transaction, { amount }));
```

Comment:
ideally, we should start using more the "updateTransaction" that comes out of the useBridgeTransaction, like in this code: https://github.com/LedgerHQ/ledger-live-desktop/blob/28ce3df1b9736e15488ca61a7a7284b6005350c1/src/renderer/families/cosmos/RedelegationFlowModal/steps/StepDestinationValidators.js#L28 ( assuming you pass it as in https://github.com/LedgerHQ/ledger-live-desktop/blob/d17038c4f175ca019935522a93adcb4066c0d45b/src/renderer/families/cosmos/UndelegationFlowModal/Body.js#L166
(setTransaction/onChangeTransaction patterns will eventually all need to be replaced to do this because atomic updater avoid race conditions)